### ISSUE TITLE
Add a "get now" method to blocking in async detection

### ISSUE THREAD

**Issue by alecgrieser** [FoundationDB/fdb-record-layer#641]
There are some instances where we have futures that we are pretty sure are done that we then wait on in asynchronous contexts. Something like:

```java
CompletableFuture<V1> f1 = someFuture();
CompletableFuture<V2> f2 = someOtherFuture();
CompletableFuture<V3> f3 = yetAnotherFuture();
return CompletableFuture.allOf(f1, f2, f3).thenApply(vignore -> {
    V1 v1 = f1.join();
    V2 v2 = f2.join();
    V3 v3 = f3.join();
    return someThingThatCombines(v1, v2, v3);
});
```

In this instance, because `allOf` will wait on `f1`, `f2`, and `f3`, the joins within the async lambda are definitely fine, and in this case, it's kind of obvious. The problem is that there are other instances where the side effect of one function might necessarily lead to one future being completed...until some refactoring makes that not the case. So it would be nice to have a way of "joining a future but don't ever block". The `.getNow` method on completable futures is *almost* what we want, but it supplies a default value instead of throwing an error on failure.

You could also imagine that you'd want different modes for this:

1. Throw an error if blocking required.
1. Block but warn.

This is somewhat analogous to the existing options in the async detection stuff, except that "waiting" on a completed future should always be allowed. (This is why using `FDBDatabase.join` isn't quite sufficient.) It's possible we can re-use some of the existing options but just tell it not to worry about "waiting" on a joined future (somehow).
